src/patches.rs

```diff
@@ -12529,13 +12645,25 @@ fn build_and_run_patches(gc_disc: &mut structs::GcDisc, config: &PatchConfig, ve
                     )
                 );
 
+                patcher.add_resource_patch(
+                    (&[pak_name.as_bytes()], room_info.mapa_id.to_u32(), FourCC::from_bytes(b"MAPA")),
+                    move |res| add_pickups_to_mapa(
+                            res,
+                            room_idx,
+                            idx,
+                            &pickup2,
+                            (*pickup_location).position
+                        )
+                );
+
                 idx = idx + 1;
                 seed = seed + 1;
             }
 
             // Patch extra item locations
             while idx < pickups_config_len {
                 let pickup = pickups[idx].clone(); // TODO: cloning is suboptimal
+                let pickup2 = pickups[idx].clone();
```

Comment:
Why is pickup2 needed? Can you maybe give it a better name?

Reply:
rust would complain of the reference to pickup being already used by modify_pickups_in_mrea and patch_add_item

Reply:
about the name I don't know what to name it tbh

Reply:
What do you mean cloning it in those functions? The reference can't be the same as the one of add_pickups_to_mapa

Try to use only pickup in both modify_pickups_in_mrea and add_pickups_to_mapa and read the error message